app/actions.go

```diff
@@ -694,3 +696,124 @@ var rebuildRoutes = action.Action{
 		return nil, nil
 	},
 }
+
+var checkSingleCNameExists = action.Action{
+	Name: "cname-exists",
+	Forward: func(ctx action.FWContext) (action.Result, error) {
+		cname := ctx.Params[1].(string)
+
+		collection, err := storagev2.AppsCollection()
+		if err != nil {
+			return nil, err
+		}
+
+		cs, err := collection.CountDocuments(ctx.Context, mongoBSON.M{"cname": cname})
+		if err != nil {
+			return nil, err
+		}
+		if cs == 0 {
+			return nil, ErrCNameDoesNotExist
+		}
+
+		return cname, nil
+	},
+}
+
+var saveCertIssuer = action.Action{
+	Name: "save-cert-issuer",
+	Forward: func(ctx action.FWContext) (action.Result, error) {
+		app := ctx.Params[0].(*App)
+		cname := ctx.Params[1].(string)
+		issuer := ctx.Params[2].(string)
+
+		var conn *db.Storage
+		conn, err := db.Conn()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+
+		sanitizedCName := strings.ReplaceAll(cname, ".", "_dot_")
+		certIssuerCName := fmt.Sprintf("certissuers.%s", sanitizedCName)
+
+		err = conn.Apps().Update(
+			bson.M{"name": app.Name},
+			bson.M{"$set": bson.M{certIssuerCName: issuer}},
+		)
+		return cname, err
+	},
+	Backward: func(ctx action.BWContext) {
+		app := ctx.Params[0].(*App)
+		cname := ctx.Params[1].(string)
+
+		var conn *db.Storage
+		conn, err := db.Conn()
+		if err != nil {
+			log.Errorf("BACKWARD remove certissuer db. unable to connect: %s", err)
+			return
+		}
+		defer conn.Close()
+
+		sanitizedCName := strings.ReplaceAll(cname, ".", "_dot_")
+		certIssuerCName := fmt.Sprintf("certissuers.%s", sanitizedCName)
+
+		err = conn.Apps().Update(
```

Comment:
@gvicentin we could rebase to update using new mongo driver.

Reply:
Thanks! Fixed ✅